### PR TITLE
Add xml-model lines as an option

### LINKED ISSUE #18: Add xml-model PI to README?
# Problem

Issue #17 gave me no peace of mind. :wink: The issue described the `xml.fileAssociations` option in the settings to associate a file extension with a (RNG) schema.

However, this has some drawbacks:

* There is no exact 1:1 mapping from XML to a specific (RNG) schema. DocBook files can be any extension. Same with DITA.
* Different (XML) files need different schemas. You can't validate a DITA XML file with the DocBook schema. If both have the same file extension, it will pick up the wrong schema.
* The setting is more or less a _convention_. Although this is per se not bad, but it's not really general enough and works only for _our_ specific use case.

# Proposal

I think we can leave the `xml.fileAssociations` as it is. However, I would propose to extend the README. For this we need to test it first. If it works, we could add it to the text. If not, no harm is done and we can close this issue. :wink: 

I'd suggest to investigate a bit time to look into the `<?xml-model?>` processing instruction (PI). It should be supported by VSCode now. This PI is included into the header of a XML file and acts like a reference. It points to a specific schema to be used for validation similar like a DOCTYPE for a DTD.

For RNG, it can be used in two ways:

* PI for RELAX NG compact schema:

       <?xml-model href="URI" type="application/relax-ng-compact-syntax"?>

* PI for a RELAX NG schema in XML format:

       <?xml-model href="URI"
                   type="application/xml"
                   schematypens="http://relaxng.org/ns/structure/1.0"?>

@tbazant: would you like to test a file with the above PIs and see if it gets validated?

# References

* [All xml-model issues from the upstream project](https://github.com/eclipse/lemminx/issues?q=label%3Axml-model+is%3Aclosed)
* [W3C: Associating Schemas with XML documents 1.0 (Third Edition)](https://www.w3.org/TR/xml-model/)

### PR DESCRIPTION
**Pull request by tbazant**
fixes #18

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -25,24 +25,41 @@ via GitHub pull requests or issues.
...
+        ```xml
+        <?xml-model href="urn:x-suse:rng:v2:geekodoc-flat"
+            type="application/xml"
+            schematypens="http://relaxng.org/ns/structure/1.0"?>
+        ```
+    - for DocBook assmbly files:
+        ```xml
+        <?xml-model href="https://cdn.docbook.org/schema/5.2/rng/assemblyxi.rng"
+            type="application/xml"
+            schematypens="http://relaxng.org/ns/structure/1.0"?>
```
Comment (tomschr) [inline README.md:44]: There is a typo in "assmbly". :wink:

I would amend it with an additional entry for DocBook files. That gives us the three important schemas we need (GeekoDoc, DocBook, and DocBook Assemblies):

```suggestion
    - for DocBook assembly files:
        ```xml
        <?xml-model href="https://cdn.docbook.org/schema/5.2/rng/assemblyxi.rng"
            type="application/xml"
            schematypens="http://relaxng.org/ns/structure/1.0"?>

    - for DocBook files:
        ```xml
        <?xml-model href="https://cdn.docbook.org/schema/5.2/rng/docbookxi.rng"
            type="application/xml"
            schematypens="http://relaxng.org/ns/structure/1.0"?>

```

File: README.md [hunk 2]
```diff
@@ -25,24 +25,41 @@ via GitHub pull requests or issues.
...
-        "systemId": "file:///usr/share/xml/geekodoc/rng/geekodoc-latest-flat.rng"
-    }
-],
-```
+### Support for RELAX NG schema
+
+To let VSCode be aware of GeekoDoc and DocBook assembly schema, you have two options:
+
+- Insert an XML model definition at the top of the XML file *before* the root
+  element (for example, after `<?xml version="1.0"/>`).
```
Comment (tomschr) [inline README.md:33]:
```suggestion
  element, but after `<?xml version="1.0"/>`.
```

File: README.md [hunk 3]
```diff
@@ -25,24 +25,41 @@ via GitHub pull requests or issues.
...
-        "pattern": "**/*.xml",
-        "systemId": "file:///usr/share/xml/geekodoc/rng/geekodoc-latest-flat.rng"
-    }
-],
-```
+### Support for RELAX NG schema
+
+To let VSCode be aware of GeekoDoc and DocBook assembly schema, you have two options:
+
+- Insert an XML model definition at the top of the XML file *before* the root
```
Comment (tomschr) [inline README.md:32]: Some comments/questions:

* Should we list the XML file extensions **before** the XML model definition? I think, the first is easier to use and should be the preferred method, thus having it listed as the first item.
* Should we recommend the XML file extensions over the XML model? I think so. A "this is the preferred method" or "recommended" after the first sentence would probably be sufficient.
* Mention that the XML model definition is only needed, if you want/need to deviate from the preferred config with the XML file extension.

**Review by tomschr — CHANGES_REQUESTED** [review]
@tbazant Thanks Tomas! Well done! :+1: I have some small additions and questions.

File: README.md (hunk 3 above)
Comment (tbazant) [reply README.md:32]: i had the feeling that xml-model is a cleaner solution, isn't it?
I can revert the order if you think it makes more sense of course :-)

Comment (tbazant) [reply README.md]: Done.

**Pull request merged**